api/envoy/api/v3alpha/core/health_check.proto

```diff
@@ -291,4 +291,11 @@ message HealthCheck {
   // initial health check failure event will be logged.
   // The default value is false.
   bool always_log_health_check_failures = 19;
+
+  // Specifies the ALPN protocol for health check connections. This is useful if the corresponding
+  // upstream is using ALPN-based :ref:`FilterChainMatch
+  // <envoy_api_msg_api.v3alpha.listener.FilterChainMatch>` along with different protocols for
+  // health checks versus data connections. If empty, no ALPN protocol will be set on HC
+  // connections.
+  string alpn_protocol = 21;
```

Comment:
WDYT about using `repeated string` for consistency with `CommonTlsContext.alpn_protocols`?
https://github.com/envoyproxy/envoy/blob/dfe687d49574ef7eb1bf84867bf571e805a2bf97/api/envoy/api/v2/auth/cert.proto#L334

Reply:
Yeah good idea.